python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -492,40 +564,45 @@ def compute_missing_subslice(
         if self.asset_graph.get(asset_key).is_materializable:
             # cheap call which takes advantage of the partition status cache
             materialized_subset = self._queryer.get_materialized_asset_subset(asset_key=asset_key)
-            materialized_slice = self.get_asset_slice_from_valid_subset(materialized_subset)
+            materialized_slice = AssetSlice(self, _AssetSliceCompatibleSubset(materialized_subset))
             return from_slice.compute_difference(materialized_slice)
         else:
             # more expensive call
             missing_asset_partitions = {
                 ap
-                for ap in from_slice.convert_to_valid_asset_subset().asset_partitions
+                for ap in from_slice.expensively_compute_asset_partitions()
                 if not self._queryer.asset_partition_has_materialization_or_observation(ap)
             }
-            missing_subset = ValidAssetSubset.from_asset_partitions_set(
-                asset_key, self._get_partitions_def(asset_key), missing_asset_partitions
-            )
-            return self.get_asset_slice_from_valid_subset(missing_subset)
+            return self.get_asset_slice_from_asset_partitions(missing_asset_partitions)
 
     @cached_method
     def compute_in_progress_asset_slice(self, *, asset_key: "AssetKey") -> "AssetSlice":
-        return _slice_from_valid_subset(
-            self, self._queryer.get_in_progress_asset_subset(asset_key=asset_key)
+        return AssetSlice(
+            self,
+            _AssetSliceCompatibleSubset(
```

Comment:
Instead of using the `slice_from_valid_subset` method, we just create the slice directly from an `AssetSubset` because we know that these methods will always return a valid subset.